### PR TITLE
Add compatibility to dynamically login to organisations from the guardio application

### PR DESCRIPTION
**Pull request by Achintha444**
## Purpose

- There was a restriction in the guardio sample application where we had to manually add the organisation's that we wish to access to the app's config.json file. 
- But after this fix, it will no longer be necessary.

Since this feature has not being push to the products, we have still kept the compatibility with SAMPLE_ORGS section in the config.json file.

### CODE REVIEW

File: b2b-sample/guardio-insurance-sample-application/pages/api/settings/switchOrg.js [hunk 1]
```diff
@@ -65,7 +65,14 @@ function getSwitchEndpoint() {
         return `${config.WSO2IS_HOST}/oauth2/token`
     }
 
-    return `${config.WSO2IS_HOST}/o/${config.WSO2IS_TENANT_NAME}/oauth2/token`
+    if(config.APP_IN_ASGARDEO_FIRST_LEVEL) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/pages/api/settings/switchOrg.js:69]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/pages/api/settings/switchOrg.js [hunk 2]
```diff
@@ -65,7 +65,14 @@ function getSwitchEndpoint() {
...
+        return `${config.WSO2IS_HOST}/t/${config.WSO2IS_TENANT_NAME}/oauth2/token`;
+    } else {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/pages/api/settings/switchOrg.js:72]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/pages/o/[id].js [hunk 3]
```diff
@@ -16,40 +16,50 @@
...
+		if (props.routerQuery) {
+			orgSignin(props.routerQuery);
+			return;
+		}
+	}, [props.routerQuery]);
+
+	return (
+		props.session
+			? <Settings orgId={props.session.orgId} name={props.session.orgName} session={props.session}
+				colorTheme={'blue'} />
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/pages/o/[id].js:62]:
```suggestion
			? <Settings 
			        orgId={props.session.orgId} 
			        name={props.session.orgName} 
			        session={props. Session}
				colorTheme={'blue'} 
			/>
```
```suggestion
			? <Settings 
			        orgId={props.session.orgId} 
			        name={props.session.orgName} 
			        session={props. Session}
				colorTheme={'blue'} 
			/>
```

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/application/callListCurrentApplication.js [hunk 4]
```diff
@@ -16,25 +16,22 @@
...
-     } catch (err) {
-        
-         return null;
-     }
- }
+import config from '../../../../config.json';
+import { getInternalApiRequestOptions } from '../../../util/apiUtil/getInteralApiRequestOptions';
+
+export default async function callListCurrentApplication(session) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/application/callListCurrentApplication.js:23]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/application/callListCurrentApplication.js [hunk 5]
```diff
@@ -16,25 +16,22 @@
...
+        const res = await fetch(
+            `${config.WSO2IS_CLIENT_URL}/api/settings/application/listCurrentApplication`,
+            getInternalApiRequestOptions(session)
+        );
+
+        const data = await res.json();
+
+        return data;
+    } catch (err) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/application/callListCurrentApplication.js:34]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callEditUser.js [hunk 6]
```diff
@@ -16,24 +16,21 @@
...
-            getInternalApiRequestOptionsWithParam(session, subOrgId, user)
+            getInternalApiRequestOptionsWithParam(session, user)
         );
 
         const data = await res.json();
 
         return data;
     } catch (err) {
-        
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callEditUser.js:33]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callPatchGeneralSettingsIdp.js [hunk 7]
```diff
@@ -16,26 +16,22 @@
...
+        const res = await fetch(
+            `${config.WSO2IS_CLIENT_URL}/api/settings/identityProvider/patchGeneralSettingsIdp/${idpId}`,
+            getInternalApiRequestOptionsWithParam(session, body)
+        );
+
+        const data = await res.json();
+
+        return data;
+    } catch (err) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callPatchGeneralSettingsIdp.js:34]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js [hunk 8]
```diff
@@ -19,25 +19,31 @@
 import callSwitchOrg from "../../apiCall/settings/callSwitchOrg";
 import { commonDecode } from "../../util/apiUtil/commonDecode";
 import { parseCookies } from '../../util/routerUtil/routerUtil';
+import config from '../../../config.json';
 
-function getSubOrgId(request) {
-    const cookies = parseCookies(request);
-    const subOrgId = cookies.orgId;
+function getSubOrgId(token) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js:25]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js [hunk 9]
```diff
@@ -19,25 +19,31 @@
...
 import { parseCookies } from '../../util/routerUtil/routerUtil';
+import config from '../../../config.json';
 
-function getSubOrgId(request) {
-    const cookies = parseCookies(request);
-    const subOrgId = cookies.orgId;
+function getSubOrgId(token) {
+    
+    try {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js:27]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js [hunk 10]
```diff
@@ -19,25 +19,31 @@
...
+        return config.SAMPLE_ORGS[0].id;
+    } catch (error) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js:30]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js [hunk 11]
```diff
@@ -19,25 +19,31 @@
...
+    const subOrgId = getSubOrgId(token);
     const accessToken = token.accessToken;
 
     try {
         const orgSession = await commonDecode(()=> callSwitchOrg(subOrgId, accessToken), null);
 
         return orgSession;
     } catch (err) {
-        
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/decodeSwitchOrg.js:46]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeUser.js [hunk 12]
```diff
@@ -18,16 +18,16 @@
 
 export default function decodeUser(user) {
 
-    if (user.id === undefined || user.userName === undefined || user.name === undefined || user.emails === undefined) {
+    if ( !user.id || !user.userName || !user.name  || !user.emails ) {
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeUser.js:22]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js [hunk 13]
```diff
@@ -64,6 +68,28 @@ function getLoggedUserId(token) {
     return parseJwt(token).sub;
 }
 
+function getOrgId(token) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js:72]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js [hunk 14]
```diff
@@ -64,6 +68,28 @@ function getLoggedUserId(token) {
...
+        return config.SAMPLE_ORGS[0].id;
+    } catch (error) {
+
+        return parseJwt(token).org_id;
+    }
+}
+
+function getOrgName(token) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js:83]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js [hunk 15]
```diff
@@ -64,6 +68,28 @@ function getLoggedUserId(token) {
...
+    try {
+
+        return config.SAMPLE_ORGS[0].name;
+    } catch (error) {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js:88]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js [hunk 16]
```diff
@@ -64,6 +68,28 @@ function getLoggedUserId(token) {
...
+        return parseJwt(token).org_id;
+    }
+}
+
+function getOrgName(token) {
+
+    try {
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/util/routerUtil/routerUtil.js:85]:
```suggestion
```

**Review by thivi — APPROVED** [review]

**Review by thivi — APPROVED** [review]

File: b2b-sample/guardio-insurance-sample-application/README.md [hunk 17]
```diff
@@ -17,3 +17,9 @@ yarn install
 yarn dev
 ```
 Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+## Additionally
+* If the correct organisation id and organisation name are returned in the idToken when a user is logging into an organisation, remove the `SAMPLE ORGS` section from the <i>config.json</i> file.
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/README.md:22]: Suggestion

```
"If the correct organization ID and organization name are returned in the IdToken when a user logs into an organization, remove the SAMPLE_ORGS section from the <i>config.json</i> file.
```

File: b2b-sample/guardio-insurance-sample-application/README.md [hunk 18]
```diff
@@ -17,3 +17,9 @@ yarn install
...
+
+#### If you are using the app with [Asgardeo](https://wso2.com/asgardeo/)
+* Please change the value of the <i>config.json</i>  file's `APP IN ASGARDEO FIRST LEVEL` to `true` if the organisation that contains the application is in the first level.
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/README.md:25]: `organization`

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js [hunk 19]
```diff
@@ -291,9 +291,9 @@ const FacebookIdentityProvider = ({ onFormValuesChange, formValues }) => {
     return (
         <Form onChange={onFormValuesChange} formValue={formValues}>
             <Form.Group controlId="application_name">
-                <Form.ControlLabel>Application Name</Form.ControlLabel>
+                <Form.ControlLabel>Idp Name</Form.ControlLabel>
                 <Form.Control name="application_name" />
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js:295]: Why this form control name leave as application_name?

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callListAllIdentityProviders.js [hunk 20]
```diff
@@ -16,25 +16,22 @@
...
-            getInternalApiRequestOptions(session, subOrgId)
+            getInternalApiRequestOptions(session)
         );
 
         const data = await res.json();
 
         return data;
     } catch (err) {
-        
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callListAllIdentityProviders.js:34]: Do we need these new lines?

File: b2b-sample/guardio-insurance-sample-application/README.md (hunk 17 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/README.md:22]: These have been fixed with the new README.md file

### File
- https://github.com/wso2/samples-is/tree/master/b2b-sample#readme

File: b2b-sample/guardio-insurance-sample-application/README.md (hunk 18 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/README.md:25]: These have been fixed with the new README.md file

### File
- https://github.com/wso2/samples-is/tree/master/b2b-sample#readme

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callListAllIdentityProviders.js (hunk 20 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callListAllIdentityProviders.js:34]: This have been fixed when the functionality of this moved to a new file

### File
- https://github.com/wso2/samples-is/blob/master/b2b-sample/libs/business-admin-app/data-access/data-access-controller/src/lib/controller/identityProvider/controllerListAllIdentityProviders/controllerCallListAllIdentityProviders.ts

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js (hunk 19 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js:295]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/75f1d1f961472868e3f94fe91c6db88ff55c2644

**Pull request merged**